code.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    while i < self.length { |i| yield(self[i]) }
+  end
+
+  def my_each_with_index
+    while i < self.length { |i| yield(self[i], i) }
```

Comment:
See the above comments on using `while` and returning `Enumerator` when no block is given.

See ruby doc on [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index) behavior.